WooCommerce/Classes/ViewRelated/Dashboard/Settings/Settings/SettingsViewModel.swift

```diff
@@ -0,0 +1,350 @@
+import UIKit
+import Combine
+import Yosemite
+import Storage
+import class Networking.UserAgent
+
+protocol SettingsViewModelOutput {
+    typealias Section = SettingsViewController.Section
+
+    /// Table view sections.
+    ///
+    var sections: [Section] { get }
+
+    /// Announcement for the current app version
+    ///
+    var announcement: Yosemite.Announcement? { get }
+
+    /// Main Account's displayName
+    ///
+    var accountName: String { get }
+
+    /// Main Site's Name
+    ///
+    var siteName: String? { get }
+
+    /// Main Site's URL
+    ///
+    var siteUrl: String? { get }
+}
+
+protocol SettingsViewModelActionsHandler {
+    /// Sets up the view model and loads the settings.
+    /// Presenter (SettingsViewController in this case) is responsible for calling this from viewDidLoad method.
+    ///
+    func onViewDidLoad()
+
+    /// Reloads the sites when store picker gets dismissed.
+    /// Presenter (SettingsViewController in this case) is responsible for calling this method when store picker is dismissed.
+    ///
+    func onStorePickerDismiss()
+}
+
+protocol SettingsViewModelInput {
+    var presenter: SettingsViewPresenter? { get set }
+}
+
+final class SettingsViewModel: SettingsViewModelOutput, SettingsViewModelActionsHandler, SettingsViewModelInput {
+
+    typealias Row = SettingsViewController.Row
+
+    weak var presenter: SettingsViewPresenter?
+
+    /// Table Sections to be rendered
+    ///
+    var sections: [Section] = []
```

Comment:
Nit: I know this variable is defined as read-only on `SettingsViewModelOutput`, but how about adding `private(set)` for extra safety?

Reply:
Great point. Changed in https://github.com/woocommerce/woocommerce-ios/pull/5393/commits/ea1f0951950a17c3bf8097032ce6ba13faf37342